### PR TITLE
New Visibility Mode: Show waypoints around the current Waypoint (cleaned version)

### PR DESCRIPTION
**Pull request by tn4799**
Cleaned up version of #2110 .

Also contains a couple of formating changes because trim trailing whitespaces is enabled.

### CODE REVIEW

File: .github/scripts/update-contributors/contributors.json [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"main":["pvaiko","Tensuko","schwiti6190"],"languages":{"en":"English","de":"German","jp":"Japanese","pl":"Polish","cz":"Czech","fr":"French","es":"Spanish","ru":"Russian","it":"Italian","pt":"Portuguese","hu":"Hungarian","nl":"Dutch","cs":"Chinese Simplified","ct":"Chinese Traditional","br":"Brazilian Portuguese","tr":"Turkish","ro":"Romanian","kr":"Korean","ea":"Spanish LA","da":"Danish","fi":"Finnish","no":"Norwegian","sv":"Swedish","fc":"Canadian French"},"translators":{"cz":["SniperKittenCZ","chroja","tn4799"],"da":["o0bonde","tn4799"],"de":["RealNeo80","Jan2903","tn4799"],"en":["tn4799"],"es":["KCHARRO","aorek","tn4799"],"fr":["Lactic68","tn4799"],"hu":["tn4799"],"it":["Roby1164","tn4799"],"ip":[],"nl":["Thodor12","tn4799"],"pl":["KITT3000","tn4799"],"pt":["Admilson","tn4799"],"ru":["Gonimy-Vetrom","tn4799"],"sl":[],"sv":["piplarsson","tn4799"],"cs":["kgdtuu","tn4799"],"br":["tn4799"],"ct":["tn4799"],"ea":["tn4799"],"fc":["tn4799"],"fi":["tn4799"],"jp":["tn4799"],"kr":["tn4799"],"no":["tn4799"],"ro":["tn4799"],"tr":["tn4799"]}}
```
Comment (pvaiko) [inline .github/scripts/update-contributors/contributors.json:1]: Please revert the changes in this file.

File: Contributors.md [hunk 2]
```diff
@@ -5,18 +5,30 @@
...
-* Dutch: [Thodor12](https://github.com/Thodor12)
-* French: [Lactic68](https://github.com/Lactic68)
-* German: [Jan2903](https://github.com/Jan2903), [RealNeo80](https://github.com/RealNeo80)
-* Italian: [Roby1164](https://github.com/Roby1164)
-* Polish: [KITT3000](https://github.com/KITT3000)
-* Portuguese: [Admilson](https://github.com/Admilson)
-* Russian: [Gonimy-Vetrom](https://github.com/Gonimy-Vetrom)
-* Spanish: [KCHARRO](https://github.com/KCHARRO), [aorek](https://github.com/aorek)
-* Swedish: [piplarsson](https://github.com/piplarsson)
+* Brazilian Portuguese: [tn4799](https://github.com/tn4799)
```
Comment (pvaiko) [inline Contributors.md:8]: Please revert the changes in this file.

File: config/VehicleSettingsSetup.xml [hunk 3]
```diff
@@ -12,11 +12,13 @@
...
 				<Value name="SHOW_COURSE_DEACTIVATED">0</Value>
 				<Value name="SHOW_COURSE_START_STOP">1</Value>
-				<Value name="SHOW_COURSE_ALL">2</Value>
+				<Value name="SHOW_COURSE_CURRENT_WPS">2</Value>
+				<Value name="SHOW_COURSE_ALL">3</Value>
 			</Values>
 			<Texts>
 				<Text prefix="false">CP_deactivated</Text>
 				<Text>startStop</Text>
+				<Text>currentWaypoint</Text>
```
Comment (pvaiko) [inline config/VehicleSettingsSetup.xml:21]: This still seems to be missing from the MasterTranslations.xml file.

**Review by pvaiko — CHANGES_REQUESTED** [review]

File: scripts/specializations/CpCourseManager.lua [hunk 4]
```diff
@@ -257,57 +258,75 @@ function CpCourseManager:getOffsetFieldWorkCourse()
...
     return spec.courses
 end
 
 function CpCourseManager:hasCourse()
-    local spec = self.spec_cpCourseManager 
+    local spec = self.spec_cpCourseManager
     return next(spec.courses) ~= nil
 end
 
+function CpCourseManager:calculateCourseVisibility()
```
Comment (schwiti6190) [inline scripts/specializations/CpCourseManager.lua:270]: @tn4799 Could you please add a ``cp`` to the function, for example like: ``cpCalculateCourseVisibility()``.
These functions are registered directly into the vehicle and every CP function should have a unique cp identifier in there for better mod compatibility and cleaner code :)

File: scripts/gui/CourseDisplay.lua [hunk 5]
```diff
@@ -259,22 +259,30 @@ function SimpleCourseDisplay:updateChangesBetween(firstIx, secondIx)
...
 	if self.course then
 		local numWp = self.course:getNumberOfWaypoints()
 		for j = 1, numWp do
-			if self.signs[j] then 
+			if self.signs[j] then
 				self.signs[j]:setVisible(visible)
-				if not self.signs[j]:isNormalSign() or j == numWp - 1 then 
-					self.signs[j]:setVisible(visible or onlyStartStopVisible)
-				end	
+				if not self.signs[j]:isNormalSign() or j == numWp - 1 then
```
Comment (schwiti6190) [inline scripts/gui/CourseDisplay.lua:268]: There are alot of contants here: ``local lowerBound = currentWaypoint - 5
						local upperBound = currentWaypoint + 20``
Might be nice to have a comment explaning the usage and also move these to global constants of the class ``SimpleCourseDisplay``

File: scripts/ai/AIDriveStrategyFieldWorkCourse.lua [hunk 6]
```diff
@@ -418,7 +418,12 @@ function AIDriveStrategyFieldWorkCourse:onWaypointPassed(ix, course)
             self:raiseImplements()
             self.state = self.states.ON_CONNECTING_TRACK
         end
-            self:checkTransitionFromConnectingTrack(ix, course)
+        self:checkTransitionFromConnectingTrack(ix, course)
+        --update lastWaypoint variable to keep moving the course when passing a new waypoint.
+        --only needs to be done when in show current waypoints mode to save performance
+        if self.vehicle:getCpSettings().showCourse:getValue() == CpVehicleSettings.SHOW_COURSE_CURRENT_WPS then
```
Comment (schwiti6190) [inline scripts/ai/AIDriveStrategyFieldWorkCourse.lua:424]: @tn4799 If I guess correctly you only want to update the changes on the lane change, when a turn between lanes happens. For cleaner code might be better to move this into onWaypointChange() and checjk if a turn is happing. 
@pvaiko What is your option and do you have an idea related to the changes mentioned above?

File: scripts/gui/CourseDisplay.lua (hunk 5 above)
Comment (tn4799) [reply scripts/gui/CourseDisplay.lua:268]: Moving them into a constant is something i've planed, but intentionally didn't do for now since my goal is to make there constants more configurable and I don't want to move them twice. 
So for now I would leave them there but add a TODO with the mention to move them to a global variable.

File: scripts/ai/AIDriveStrategyFieldWorkCourse.lua (hunk 6 above)
Comment (tn4799) [reply scripts/ai/AIDriveStrategyFieldWorkCourse.lua:424]: No, I want to update the waypoints every time the current waypoint changes so that the user can see in real time what the current and next couple of steps are. It doesn't matter if the driver turns or not.

The location of this code was suggested by @pvaiko in my old pull request.

Comment (schwiti6190) [reply scripts/ai/AIDriveStrategyFieldWorkCourse.lua:424]: @tn4799 It's fine. I mistakenly thought the code was in a different function then ``onWaypointPassed()``.

File: scripts/gui/CourseDisplay.lua (hunk 5 above)
Comment (schwiti6190) [reply scripts/gui/CourseDisplay.lua:268]: @tn4799 That's okay for now
